lib/Server.js

```diff
@@ -39,7 +39,12 @@ if (!process.env.WEBPACK_SERVE) {
 }
 
 class Server {
-  constructor(compiler, options = {}) {
+  constructor(options = {}, compiler) {
+    // TODO: remove this after plugin support is published
+    if (options.hooks) {
```

Comment:
Let's rename `options.plugin`, but will not docs it, it is internal

Reply:
I didn't get what you want to say here 😅

Reply:
Oh I see, you just check `options.hooks` to if it is exists, it is compiler, otherwise options, right?

Reply:
Yup

Reply:
Good idea